### PR TITLE
add CP2K easyconfig for psmp build

### PR DESCRIPTION
**Pull request by boegel**
@wpoely: please review/test (use https://github.com/hpcugent/easybuild-easyblocks/pull/509)

### CODE REVIEW

File: easybuild/easyconfigs/c/CP2K/CP2K-2.5.1-intel-2014b-psmp.eb [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+sources = [SOURCELOWER_TAR_BZ2]
+source_urls = [SOURCEFORGE_SOURCE]
+
+patches = [
+    'CP2K-20131211-ifort-compiler-bug-fix.patch',
+    'CP2K-2.4.0-fix_compile_date_lastsvn.patch',
+]
+
+dependencies = [
+    ('libint2', '2.0.3'),
```
Comment (boegel) [inline easybuild/easyconfigs/c/CP2K/CP2K-2.5.1-intel-2014b-psmp.eb]: why?

**Comment by boegel** [comment]
afair there were some problems with this... never managed to get back to it, maybe someone else can take closer look?

@wpoely86, @ocaisa?

**Comment by boegel** [comment]
Missing easyconfig:

```
Irresolvable dependencies encountered: libint2/2.0.3-intel-2014b
```

File: easybuild/easyconfigs/c/CP2K/CP2K-2.5.1-intel-2014b-psmp.eb (hunk 1 above)
Comment (wpoely86) [inline easybuild/easyconfigs/c/CP2K/CP2K-2.5.1-intel-2014b-psmp.eb]: see older discussion about this. Bottom line: CP2k needs the 1.x serie.

Comment (wpoely86) [inline easybuild/easyconfigs/c/CP2K/CP2K-2.5.1-intel-2014b-psmp.eb]: replace by ` ('Libint', '1.1.4'),`

**Comment by wpoely86** [comment]
replace the dep of libint and its lgtm

File: easybuild/easyconfigs/c/CP2K/CP2K-2.5.1-intel-2014b-psmp.eb [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+    ('libxc', '2.2.0'),
+    ('FFTW', '3.3.4'),
```
Comment (ocaisa) [inline easybuild/easyconfigs/c/CP2K/CP2K-2.5.1-intel-2014b-psmp.eb]: This dep isn't necessary (according to the non-smp easyconfig).

File: easybuild/easyconfigs/c/CP2K/CP2K-2.5.1-intel-2014b-psmp.eb (hunk 1 above)
Comment (ocaisa) [inline easybuild/easyconfigs/c/CP2K/CP2K-2.5.1-intel-2014b-psmp.eb]: `('Libint', '1.1.4'),`

File: easybuild/easyconfigs/c/CP2K/CP2K-2.5.1-intel-2014b-psmp.eb (hunk 2 above)
Comment (boegel) [inline easybuild/easyconfigs/c/CP2K/CP2K-2.5.1-intel-2014b-psmp.eb]: I think this is me trying to fix the problem I was running into, it had something to do with linking to FFTW...

**Comment by ocaisa** [comment]
I didn't have the impi sources so I tried a build without the FFTW dep and with this command: 
```
eb CP2K-2.5.1-intel-2014b-psmp.eb --try-toolchain-version=2015a
```
and got a successful build

**Comment by boegel** [comment]
retrying the build myself too now (with FFTW, will recheck without FFTW later)

**Comment by boegel** [comment]
@ocaisa: when using the Intel MKL that is included in `intel/2014b`, I was running into this (at runtime):

```
CP2K|  MPI error 0 in MPI library does not support the requested level of threading (MPI_THREAD_FUNNELED). : No MPI error
```

The issue that ended up blocking this PR (with FFTW included), was this (for some reason, FFTW is being picked up, or somesuch):

```
 ********************************************
 *** ERROR in init_fft (MODULE fft_tools) ***
 ********************************************
 
 *** Intel's FFTW3 interface to MKL is not thread-safe!  Please rebuild ***
 *** CP2K, linking against FFTW 3 from www.fftw.org. Now exiting...     ***
 
 *** Program stopped at line number 265 of MODULE fft_tools ***
```

**Comment by boegel** [comment]
@ocaisa: it's possible this problem is resolved in the Intel MKL used in `intel/2015a`, but the one used in `intel/2014b` (imkl 11.1.2.144) should also be fine, according to http://www.prace-ri.eu/IMG/pdf/wp140.pdf (see page 10, section on "Fast Fourier Transforms")

**Comment by ocaisa** [comment]
Since CP2K is the one flagging the error, then maybe its because the MKL was fixed but the error capture in this version of CP2K isn't capable of picking that up?

**Comment by boegel** [comment]
@ocaisa: I think I checked that back then, but I can look into it again...

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in this PR)
Linux SL 6.7, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.6.6
See https://gist.github.com/2ea6ed1b7f0cbe848aab for a full test report.

**Comment by boegel** [comment]
@ocaisa: last build worked, but my log is filled with these (for the CP2K regtest):

```
     ch4-ch4-gpw-vdw.inp                                        RUNTIME FAIL ! 
```

and

```
CP2K|  MPI error 0 in MPI library does not support the requested level of threading (MPI_THREAD_FUNNELED). : No MPI error                                                                                                                                                  CP2K| Abnormal program termination, stopped by process number 0                                                                                                                                                                                                           application called MPI_Abort(MPI_COMM_WORLD, 1) - process 0                                                                                                                                                                                                                rank 0 in job 1  localhost_49167   caused collective abort of all ranks                                                                                                                                                                                                      exit status of rank 0: return code 1                                                                                                                                                                                                                                    EXIT CODE:  1  MEANING:  RUNTIME FAIL      
```

How is the regtest doing for you? Look for `Number of FAILED` in the log...

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in this PR)
Linux SL 6.6, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.6.6
See https://gist.github.com/d3b1c0e90bef1a86f4f3 for a full test report.

**Comment by ocaisa** [comment]
This was a bad PR choice for me, we use custom slurm and mpirun doesn't execute properly for impi, you must use srun (this is a general problem we have when EB tests require this). Short story shorter, number of failed tests is 2469.

**Comment by boegel** [comment]
@ocaisa: ah, ok... I'll give it a shot myself with `2015a` (or `2015b`)

**Comment by JensTimmerman** [comment]
jenkins test this please

**Comment by boegel** [comment]
@wpoely86: up for taking a look at this?

**Comment by wpoely86** [comment]
Sure, if we got a working build system again :wink:

**Comment by boegel** [comment]
@wpoely86: ping? :)

**Comment by wpoely86** [comment]
We have a working psmp build of CP2K in #2889

**Pull request closed without merging**